aten/src/ATen/native/quantized/cpu/qlinear.cpp

```diff
@@ -797,14 +797,20 @@ at::Tensor PackedLinearWeightsOnednn::apply_impl(
   TORCH_CHECK(
       dim != 0,
       "qlinear (ONEDNN): input dim should be at least 1, but got 0");
-  TORCH_CHECK(input.scalar_type() == c10::ScalarType::QUInt8,
-      "qlinear (ONEDNN): data type of input should be QUint8.");
+  TORCH_CHECK(input.scalar_type() == c10::ScalarType::QUInt8 || input.scalar_type() == c10::ScalarType::QInt8,
+      "qlinear (ONEDNN): data type of input should be QUInt8 or QInt8.");
+
+#ifdef __aarch64__
+  using qtype = int8_t;
```

Comment:
shouldn't this be derived from the input dtype itself?

Reply:
Done.